### PR TITLE
azdev scaffolding commands and setup tweaks

### LINKED ISSUE #18: Add command to generate module or extension template
`azdev create -n foo -t {module|extension}`

This command would generate all of the boilerplate code to create a CLI module or extension. This avoids the pitfalls of people trying to copy-paste an existing example or rely on documentation that is out-of-date. The scaffolded files will have examples for how to do common things and we can use this to enforce best practices.

### LINKED ISSUE #63: Paths do not "expand user" in interactive setup
During interactive setup, passing ~ to indicate the home directory does not expand. This works correct when specifying via the command's options.

### LINKED ISSUE #69: Expose pip arguments for azdev setup
It would be useful to add options to `azdev setup` that are the equivalent of pip options `--no-cache-dir` and `--ignore-installed`. Currently, there is no way to apply those.

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #18. Fixes #63.

### CODE REVIEW

**Comment by tjprescott** [comment]
@zikalino @achandmsft here is the PR for the `cli/extension create` commands.

**Comment by zikalino** [comment]
@tjprescott thank you! it's pretty large, I will put some comments very soon.
I am actually wondering how we could do more with some additional information from swagger.

**Comment by tjprescott** [comment]
@zikalino we were planning to more seriously investigate auto-generation of commands from Swagger but decided this has value in its own right and is pretty low-cost.

I will likely add an interactive mode for this command (similar to `azdev setup`) to better walk newcomers through the process.

**Comment by zikalino** [comment]
@tjprescott I agree, I think we should definitely merge as it definitely creates very good framework that can be extended.

File: azdev/help.py [hunk 1]
```diff
@@ -225,17 +254,3 @@
     short-summary: >
         List the repositories that will be searched for in-development extensions.
 """
-
```
Comment (zikalino) [inline azdev/help.py]: that was something what was removed before, right?

File: azdev/mod_templates/_client_factory.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+def cf_{{ name }}(cli_ctx, *_):
+{% if client_name %}
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from {{ sdk_path }} import {{ client_name }}
+    return get_mgmt_service_client(cli_ctx, {{ client_name }})
+{% else %}
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    # TODO: Replace CONTOSO with the appropriate label and uncomment
+    # from azure.mgmt.CONTOSO import CONTOSOManagementClient
```
Comment (zikalino) [inline azdev/mod_templates/_client_factory.py]: Shouldn't this be replaced by the value passed via --client-name?

**Comment by tjprescott** [comment]
Thanks @zikalino. I have a few final points that will add the right annotations to the CODEOWNERS file and some other little housekeeping items, but I hope to merge and release today or tomorrow.

**Comment by zikalino** [comment]
@tjprescott I forgot to mention, I used your branch and have tested it and it works for me!

**Comment by zikalino** [comment]
I forgot to mention one more thing.
I have generated apimanagement command, and there was one exception, to fix it I had to make change in one of the autogenerated files:

/workspaces/azure-cli/src/command_modules/azure-cli-apimanagement/azure/cli/command_modules/apimanagement/commands.py

Just at the end of file i removed **is_preview=True** from following code:

    with self.command_group('apimanagement', is_preview=True):
        pass

The exception was:

The command failed with an unexpected error. Here is the traceback:

unrecognized kwargs: ['is_preview']
Traceback (most recent call last):
  File "/azure-cli/env/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/workspaces/azure-cli/src/azure-cli-core/azure/cli/core/commands/__init__.py", line 439, in execute
    self.commands_loader.load_command_table(args)
  File "/workspaces/azure-cli/src/azure-cli-core/azure/cli/core/__init__.py", line 247, in load_command_table
    _update_command_table_from_modules(args)
  File "/workspaces/azure-cli/src/azure-cli-core/azure/cli/core/__init__.py", line 144, in _update_command_table_from_modules
    module_command_table, module_group_table = _load_module_command_loader(self, args, mod)
  File "/workspaces/azure-cli/src/azure-cli-core/azure/cli/core/commands/__init__.py", line 893, in _load_module_command_loader
    return _load_command_loader(loader, args, mod, 'azure.cli.command_modules.')
  File "/workspaces/azure-cli/src/azure-cli-core/azure/cli/core/commands/__init__.py", line 875, in _load_command_loader
    command_table = command_loader.load_command_table(args)
  File "/workspaces/azure-cli/src/command_modules/azure-cli-apimanagement/azure/cli/command_modules/apimanagement/__init__.py", line 24, in load_command_table
    load_command_table(self, args)
  File "/workspaces/azure-cli/src/command_modules/azure-cli-apimanagement/azure/cli/command_modules/apimanagement/commands.py", line 26, in load_command_table
    with self.command_group('apimanagement', is_preview=True):
  File "/workspaces/azure-cli/src/azure-cli-core/azure/cli/core/__init__.py", line 422, in command_group
    return self._command_group_cls(self, group_name, **kwargs)
  File "/workspaces/azure-cli/src/azure-cli-core/azure/cli/core/commands/__init__.py", line 989, in __init__
    merged_kwargs = self._merge_kwargs(kwargs, base_kwargs=command_loader.module_kwargs)
  File "/workspaces/azure-cli/src/azure-cli-core/azure/cli/core/commands/__init__.py", line 1013, in _merge_kwargs
    return _merge_kwargs(kwargs, base, CLI_COMMAND_KWARGS)
  File "/workspaces/azure-cli/src/azure-cli-core/azure/cli/core/commands/__init__.py", line 964, in _merge_kwargs
    raise TypeError('unrecognized kwargs: {}'.format(unrecognized_kwargs))
TypeError: unrecognized kwargs: ['is_preview']

**Comment by tjprescott** [comment]
@zikalino in regards to the `is_preview` kwarg, you need to have the latest bits from the dev branch. The min CLI core version for these commands is 2.0.66.

**Pull request merged**